src/enkf/readconvobs.f90

```diff
@@ -1072,6 +1073,66 @@ subroutine get_convobs_data_bin(obspath, datestring, nobs_max, nobs_maxdiag,   &
         obtype == 'sst' .or. obtype == ' rw' .or. obtype == 'dbz' .or. &
         obtype == 'gps' .or. obtype == ' dw' .or. obtype == ' pw')  then
 
+!   direct reflectivitiy DA has a different routine for dbz obs.
+     if (l_use_enkf_directZDA .and. obtype == 'dbz' ) then
+       allocate(cdiagbuf(ii),rdiagbuf(nreal,ii))
+       read(iunit) cdiagbuf(1:ii),rdiagbuf(:,1:ii)
+
+       if(twofiles)then
+          allocate(cdiagbuf2(ii2), rdiagbuf2(nreal2,ii2))
+          read(iunit2) cdiagbuf2(1:ii2),rdiagbuf2(:,1:ii2)
+       end if
+
+       do n=1,ii
+          nobdiag = nobdiag + 1
+          if(rdiagbuf(12,n) < zero .or. rdiagbuf(16,n) < errorlimit .or. &
+             rdiagbuf(16,n) > errorlimit2)cycle
+          if(abs(rdiagbuf(17,n)) > 1.e9_r_kind  .or. &
+               rdiagbuf(6,n) < 0.001_r_kind .or. &
+               rdiagbuf(6,n) > 1200._r_kind) cycle
+          if(twofiles)then
+          if(rdiagbuf(1,n) /= rdiagbuf2(1,n) .or. abs(rdiagbuf(3,n)-rdiagbuf2(3,n)) .gt. 1.e-5 .or. &
+             abs(rdiagbuf(4,n)-rdiagbuf2(4,n)) .gt. 1.e-5 .or. abs(rdiagbuf(8,n)-rdiagbuf2(8,n)) .gt. 1.e-5)then
+             write (6,*) obtype, ' conv ob data inconsistency '
+             write (6,*) (rdiagbuf(i,n),i=1,8)
+             write (6,*) (rdiagbuf2(i,n),i=1,8)
+             call stop2(-98)
```

Comment:
double check here. Do you really mean -98?

Reply:
I changed it from –98 to 94 like other stop2 routines in this code

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-2e5731fbf5a28dae90623965b46ef48217e5ffe62527725c01d71e58b01be811R1184